### PR TITLE
TSERV-782 Print Report to Console

### PR DESCRIPTION
**Pull request by bsubba**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,11 @@ Each job has a job ID, with the command "jobs cancel" a running (or queued) job
 
 `testengine jobs status <job ID>`
 
+### Print an execution report to the console'
```
Comment (lootic) [inline README.md:119]: Nice!

File: bin/jobs_functions.js [hunk 2]
```diff
@@ -44,6 +45,15 @@ module.exports = {
                 }
                 break;
             }
+            case 'printreport': {
```
Comment (lootic) [inline bin/jobs_functions.js:48]: The ticket states:
"A flag/parameter to 'printReport' can be used when submitting a request that will return a report (such as when **submitting a synchronous job** or requesting a result from a previous job)"

Not sure, but getting the feeling we miss the combination where we supply this flag together with the flag for a synchronous job.

Comment (bsubba) [reply bin/jobs_functions.js:48]: I get the same feeling now. You mean it should be possible to print the report when the job is completed if it is synchronous, right?

Comment (lootic) [reply bin/jobs_functions.js:48]: yes

Comment (bsubba) [reply bin/jobs_functions.js:48]: Now it is possible to print report when the job is completed in case of synchronous jobs

File: README.md [hunk 3]
```diff
@@ -65,6 +65,7 @@ table includes the currently implemented options.
 | Option  | Sample Value  | Description|
 |---|---|---|
 | async  | N/A  | Run the project without waiting for results. The TestJobID will be printed on the console to make it possible to query status, get reports or cancel the job|
+| printReport  | N/A  | Print the execution report to the console for synchronous job after it is completed |
```
Comment (lootic) [inline README.md:68]: I think you should explain both scenarios. Both synchronous and printing it after it has run.

File: README.md (hunk 1 above)
Comment (lootic) [reply README.md:120]: Now this comment needs to be updated with infor about running the job synchronous.

File: bin/run_functions.js [hunk 4]
```diff
@@ -454,13 +458,16 @@ function executeProject(filename, project, options) {
...
             } else {
-                if (!('async' in options)) {
+                if (!options.async || !('async' in options)) {
                     // If status isn't CANCELED, PENDING or DISCONNECTED and we have an output directory, store reports there
                     //
                     if (!missingFiles) {
                         if (config.showProgress)
                             util.output('');
                         util.output("Result: " + status);
+                        if (options.printReport) {
```
Comment (lootic) [inline bin/run_functions.js:468]: Is this running another call to the server? I think you get the report as the response text when running async.

File: README.md (hunk 3 above)
Comment (bsubba) [reply README.md:68]: This one is  when the job is run synchronously. And to print the report for already run job is different which I have written already.

File: bin/run_functions.js (hunk 4 above)
Comment (bsubba) [reply bin/run_functions.js:468]: Oh yes this is another call to server.You are right, response can be used as the report. 👍

File: README.md (hunk 3 above)
Comment (lootic) [reply README.md:68]: ah, ok!

**Review by lootic — APPROVED** [review]

**Review by brahmdev — APPROVED** [review]
looks good to me.

**Review by akkikumar72 — APPROVED** [review]

File: bin/run_functions.js [hunk 5]
```diff
@@ -377,7 +377,13 @@ function executeProject(filename, project, options) {
                             jobId = result.body['testjobId'];
                             if (config.verbose || options.async) {
                                 util.output("TestJoB ID: " + jobId);
-                            } else {
+                            }
+                            
+                            if (('printReport' in options) && options.async) {
```
Comment (lootic) [inline bin/run_functions.js:382]: Nice!

**Pull request merged**